design_documents/expect_events_cheatcode.md

```diff
@@ -0,0 +1,90 @@
+# `expect_events` Cheatcode
+
+## Context
+
+Some contract functions can emit events. It is important to test if they were emitted properly.
+
+## Goal
+
+Propose a solution that will allow checking if events were emitted.
+
+## Considered Solutions
+
+### `expect_events` Cheatcode
+
+Introduce a cheatcode with the signature:
+
+```cairo
+fn expect_events(events: Array<Event>)
```

Comment:
Does the `Event` struct already exist? If not it should be a part of the proposal as well.